Version 6.0.0-Alpha3

kabeor opened the pull request:
<!-- Filling this template is mandatory -->

**Your checklist for this pull request**
- [ ] I've documented or updated the documentation of every API function and struct this PR changes.
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)

**Detailed description**

<!-- Explain the **details** for making this change. Is a new feature implemented? What existing problem does the pull request solve? How does the pull request solve these issues? Please provide enough information so that others can review your pull request. -->

...

**Test plan**

<!-- What steps should the reviewer take to test your pull request? Demonstrate the code is solid. Or what test cases you added. Disassembly tests should be added in suite/cs_test/issues.cs -->

...

**Closing issues**

<!-- put "closes #XXXX" in your comment to auto-close the issue that your PR fixes (if any). -->

...

kabeor committed:
Update changelog for V6.0.0-Alpha1 (#2493)

* update version to v6-alpha1

* update bindings const values

* Update changelog for V6.0.0-Alpha1

Rot127 committed:
Remove irrelevant changes. (#2495)

Rot127 committed:
Fixing UB santizer, `LITBASE` and assert errors. (#2499)

thestr4ng3r committed:
AArch64: Replace vararg add_cs_detail by multiple concrete functions

Fixes UB caused by various mismatches on how these arguments are passed
and read. This became visible when running on PowerPC hosts with e.g.
`cstool -d aarch64 204862f8`.
Apart from the UB fix, this is meant to be a pure refactor.

Partially addresses #2458

Rot127 committed:
Update labeler with Xtensa and v6 files. (#2500)

Rot127 committed:
Add hard asserts to all SStream functions and memset MCInst. (#2501)

Rot127 committed:
Only trigger on released action. (#2497)

thestr4ng3r committed:
Fix cstest build with Ninja (#2506)

Rot127 committed:
Tricore EA calculation (#2504)

thestr4ng3r committed:
Update libcyaml dependency in cstest to 1.4.2 (#2508)

b1llow committed:
xtensa: Fix Branch Target (#2516)

* xtensa: Fix Branch Target

* auto-sync: fix byte pattern

* xtensa: add branch insn tests

* Revert "auto-sync: fix byte pattern"

This reverts commit cf8e870f776889514b69a2f25a376f62ab2b291c.

Rot127 committed:
Fix #2509. (#2510)

Compatibility headers should always include the header in the same dir.

Rot127 committed:
Merge pull request #2507 from thestr4ng3r/no-varargs-aarch64

AArch64: Replace vararg add_cs_detail by multiple concrete functions

Rot127 committed:
Fix stringop-truncation warning some compilers raise. (#2522)

Rot127 committed:
Add CC and VAS compatibility macros (#2525)

Rot127 committed:
Fix endianess issue during assignment. (#2528)

Rot127 committed:
This time actually fix big endian issue. (#2530)

b1llow committed:
tricore: fixes #2474 (#2523)

* tricore: fix auto-sync tricore

* tricore: fixes TriCoreGenCSMappingInsnName.inc

* tricore: fixes

* tricore: try fix ld.a SC

* tricore: fixes all

* Add TriCore to .github/workflows/auto-sync.yaml

* Add TriCore details tests(a15, d15, a10|sp)

AndrewQuijano committed:
Change CI to create Debian Package to Release (#2521)

* Updating CI to create Debian package and version is assigned by tag
version. Also updating release CI to not use end-of-life workflows

* Clear up usage of static libraries.

- Python bindings only use the dynamic lib. But built and copied the static ones sometimes nonetheless.
- Add toggles to build only static, static/dyn or only dynamic.

---------

Co-authored-by: Rot127 <unisono@quyllur.org>

Rot127 committed:
Rename build arguments: (#2534)

- BUILD_SHARED_LIBS -> CAPSTONE_BUILD_SHARED_LIBS
- BUILD_STATIC_LIBS -> CAPSTONE_BUILD_STATIC_LIBS
- BUILD_STATIC_LIBS -> CAPSTONE_BUILD_STATIC_MSVC_RUNTIME

b1llow committed:
xtensa: update to espressif/llvm-project (#2533)

b1llow committed:
fix coverity (#2546)

- cid 514642

- cid 514643

- cid 514644

- cid 514645

Rot127 committed:
Move debian package generation to a dispatch only workflow (#2543)

* Move deb package gen files int package/deb

* Fix basename check

* Make debian package generation dispatch only

Antelox committed:
Python package building rework (#2538)

* - Refactored setup.py to remove hacks regarding packaging of wheels for different platforms, improve and cleanup the code
- Updated README.txt
- Removed old Makefile and build_wheel.sh scripts
- Created a new workflow that takes care of building and testing python packages for different platforms/architectures/python versions

* Added SPDX headers to the setup.py

* - cstest_py: Fixed positional argument since it doesn't accept a `required` flag. It turns to have a mandatory tests folder path
- integration_tests.py: Use pathlib to determine the required path
- GitHub action: Simplified the tests execution command

* GitHub Actions: Run python 3.8 (lowest) and 3.13 (current highest) for native runners only during testings and the rest during tag release

* GitHub Action:
- Fixed the cibw_build matrix element
- Added a step to prepare artifact name

* GitHub Action: Added run_tests.py script to run all tests during CI workflow

* - Added SPDX headers to the run_tests.py script and to the build-wheels-publish.yml workflow file
- Minor fixes to the workflow as pointed out in the PR review
- Updated MANIFEST.in to reflect the actual libraries built during python wheel creation process
- Use subprocess.run in place of os.system in run_tests.py script

* GitHub Action:
- Run qemu step only if non-native Linux runner
- Added arch:universal2 matrix element for macos-latest runner

* Python bindings: Refreshed the list of files needed to be copied for sdist archive

* GitHub Action: Commented out arch:x86 matrix elements

* GitHub Action: Run qemu step only if non-native Linux runner

* GitHub Action: Minor fixes

* Python bindings: Added missing .in pattern when collecting src files for sdist archive

Rot127 committed:
Auto-Sync reproducability + ARM update (#2532)

b1llow committed:
fix xtensa DecodeMR23RegisterClass and add tests for MAC16 instru… (#2551)

* fix xtensa `DecodeMR23RegisterClass` and add tests for `MAC16` instructions

* revert

kabeor committed:
Prepare for update (#2552)

* Bindings(chore): Fix DeprecationWarning

* Version(upgrade): update bindings const

* Fix(chore): Fix ARMCC_Invalid is not defined

kabeor committed:
Update Changelog Version to 6.0.0-Alpha2 (#2553)

* Bindings(chore): Fix DeprecationWarning

* Version(upgrade): update bindings const

* Fix(chore): Fix ARMCC_Invalid is not defined

* Changelog: Update to version 6.0.0-Alpha2

kabeor committed:
Merge branch 'v6' into next

Rot127 committed:
PPC LLVM 18 (#2540)

* Update PPC module to LLVM 18.

**New**

(According to LLVM changelog)

- Added DFP instruction.
- Added the SCV instruction.

**Changes**

- Memory decoder were simplified by decoding disponent and base reg separately.
- `DFORM` -> `DFORM_BASE`
- Use inverted `MCInstDesc` table.
- Replace the many declared printer in PPCInstPrinter with `static inlines`.
- Renamed groups to upper case.
- Switched to `ARCH_add_cs_detail_X()` function names.
- Remove `PPCInstPrinter.h` because it is no longer used.

* Fix: Use correct directory name.

* Fix segfaults and add asserts for these NULL cases.

* Allow to map a single LLVM option to multiple CS options

* Add default endian option to the MCUpdater

* Fix setter for Little endian

* Add SPE option to cstool

* Fix QPX instructions.

Due to https://github.com/llvm/llvm-project/commit/4b43ef3e5c37459996ce0f53615881f436cb0e65
the names of the operands were matched.
Because FRT dosn't exist in the XForm_1 class,
the generated tables didn't decoded them.

* Fix: AbsAddr should be printed as unsigned.

* Fix S12 immediate printing for PC memory operands

* Fix MCUpdater tests

* Update PPCRegisterInfo_stripRegisterPrefix

* Add support for selection of Power versions

* Run clang-format

* Fix feature check

* Allow to overwrite in multi-mode

* Add some more flags

* Fix order and map name

* Add new test files.

* Fix checks for features.

Only enables PowerX feature checks of a Power architecture is enabled
and the feature is in the list of it.

* Print byte sequence with space between comma.

This helps with copy and search of the byte string in the test files.

* Fix tests broken due to feature toggles

* Shorten generated names.

* Update bindings

Rot127 committed:
Arm regressions (#2556)

* Fix: Set writeback for AddrMode5 operands with W=1

* Fix memory acccess of vector load instructions.

* Remove unused files.

* Fix operands of RFED instructions

They have now memory operand and
the writeback flag is set accordingly.

* Fix: Remove invalid mnemonic enum adr_

* Add missing NULL check

Antelox committed:
Python bindings: Enable more archs + bump cibuildwheel action to the v2.22.0 (#2558)

* GitHub Actions:
- Un-comment the matrix elements for the archs different from x86
- Bump pypa/cibuildwheel action to v2.23.0

* GitHub Action: Add concurrency option to cancel running workflows once pushing

peace-maker committed:
Add arm64 and sysz compatibility layer to Python bindings (#2559)

* Add arm64.py and sysz_const.py compatibility layer to the Python bindings

When importing `capstone.arm64` before doing other capstone imports the ARM64 constants get monkey patched into the exports.
Same with `capstone.sysz_const` for the SYSZ constants.

* Add SPDX-License comments

* Access an instruction detail in compatibility test

* Fail python tests on error instead of just ERROR output

Include the test_iter.py in the test_all.py too.

* Improve python compatibility docs

Co-authored-by: Rot127 <45763064+Rot127@users.noreply.github.com>

* Test for ARM_CC_* constants presence

---------

Co-authored-by: Rot127 <45763064+Rot127@users.noreply.github.com>

Rot127 committed:
Make thumb, v8 and m-class positional cstool arguments. (#2557)

Don't allow disabling options for now.
Will be addresses separately. Because it needs API documentation.

Co-authored-by: Wu ChenXu <kabeor00@gmail.com>

satk0 committed:
Small arm64 compat header fixes (#2563)

* Small compat header fixes

* Add ARM64_CC_NV to header patcher

Rot127 committed:
PPC regressions (#2575)

Rot127 committed:
Fix 2572 (#2574)

Rot127 committed:
Clean up the cstest documentation and build instructions. (#2580)

Roeegg2 committed:
Update BPF arch (#2568)

hainest committed:
x86: update read/write registers for transfer instructions (#2578)

hainest committed:
cstest: use DOWNLOAD_EXTRACT_TIMESTAMP conditionally (#2581)

* cstest: use DOWNLOAD_EXTRACT_TIMESTAMP conditionally

This feature was added in CMake 3.24.0, so this allows older CMakes to
still build the tests.

* Fix comparison

* Use POLICY CMP0135

AndrewQuijano committed:
Update v6 to have Debian Packages (#2579)

Roeegg2 committed:
Fix complex atomic instructions handling (#2584)

Rot127 committed:
Fix linking issue on Windows. (#2587)

Inline functions are apparently not exposed with default visibility.
Fixes: https://github.com/capstone-engine/capstone/issues/2585

Rot127 committed:
Clarify between machine used vs. Capstone module affected. (#2586)

wargio committed:
Fix LoongArch aliases and CS_OPT_SYNTAX_NO_DOLLAR support (#2594)

* Fix CS_OPT_DETAIL_REAL & CS_OPT_SYNTAX_NO_DOLLAR support in LoongArch

* Fix comment

AndrewQuijano committed:
Update Actions (#2593)

bumped the version to the latest version of some common GitHub Actions

Rot127 committed:
Downgrade labeler to v4 due to https://github.com/actions/labeler/issues/710. (#2598)

AndrewQuijano committed:
delete travis (#2600)

AndrewQuijano committed:
Create debian and rpm package on releases (#2590)

Changqing-JING committed:
Add tricore tc1.8 instructions (#2595)

* Add tricore tc1.8 instructions:
add.df
sub.df
madd.df
msub.df
mul.df
div.df
cmp.df
max.df
min.df
min.f
max.f
dftoi
dftoiz
dftoin
ftoin
dftou
dftouz
dftol
dftoul
dftoulz
abs.f
abs.df
dftolz
neg.df
neg.f
qseed.df
itodf
utodf
ltodf
ultodf
dftof
ftodf

* Fix python binding

* Fix python binding

* add tricore tc1.8 instructions
div64
div64.u
rem64
rem64.u

* add tricore tc1.8 instruction to tests/details

* Fix review

andelf committed:
Apply new EVM opcode updates (#2602)

* evm: apply new opcode updates and add test cases

* evm: use predefined invalid opcode 0xfe

* python: add CsInsn.is_invalid_insn to bypass evm issue

Changqing-JING committed:
Fix SystemZ macro in Makefile (#2603)

Rot127 committed:
Rebased #2570 (#2614)

* Add ARC files

* Added ARC files for successful compilation

* Refactor ARC files

* Add ARC c/cs tests

* Add ARC python test/bindings

* Add ARC to CI/CD

* Avoid omitting parameter names

* Update cs files

* Fix ARC bugs

* Update ARC python bindings

* Refactor and update ARC test files

* Add detail flag to arc test

* Fix ARC test problems

* Fix ARCMapping compile error

* Replace __CHAR_BIT__ to CHAR_BIT

* Add credits and ARC info

* Update ARC to match the latest next

* Python formatting

* Remove asserts on 'Unknown condition code passed'

* Add ARC to some more documentation

* Add ARC to Targets constants

* Add ARC support to llvm-tblgen

* Replace asserts & add Expr handling

* Check DecodeGPR32RegisterClass return value

* Fix fieldFromInstruction patch

* Refactor ARC

* Reformat python files

* Fix incorrect import

* Update inc files and insn names

* Update python files

* Disable AArch64 Linux wheel build due to https://github.com/capstone-engine/capstone/issues/2615.

---------

Co-authored-by: R33v0LT <sibirtsevdl@gmail.com>

kabeor committed:
chore(version): Update Version to 6.0.0-Alpha3 (#2616)

* chore(version): Update Version to 6.0.0-RC1

* chore(version): change version name to 6.0.0-alpha3

Changqing-JING committed:
Fix wrong version requirement of tricore instructions: (#2620)

crc32.b
crc32b.w
crc32l.w
crcn
popcnt.w
shuffle

Remove invalid instruction:
BISR_rc_v161

Learn up misconfigure of nor and not

Antelox committed:
Switch to ubuntu-24.04-arm runner image (#2625)

Rot127 committed:
Build Tarball before DEB/RPM package. (#2627)

Because it was run after the RPM/DEB package build it contained the 'build' directory
with all its files. Which made it way too big.

wargio committed:
Add aliases mapping for MIPS & test for id, alias_id (#2635)

wargio committed:
Add checks for MIPS details on cstest_py (#2640)

Rot127 committed:
Give the user some guidance where to add missing enumeration values. (#2639)

Rot127 commented:
Something is still off with the commit history. There shouldn't be any conflicts. Maybe the rebase was based on the wrong commit?

Antelox committed:
- Added missing files for sdist archive (#2624)

imphil committed:
cmake: Fix building capstone as sub-project (#2629)

* cmake: Use PROJECT_SOURCE_DIR instead of CMAKE_SOURCE_DIR

capstone can be built as sub-project through cmake's fetch_content
mechanism. In this case, `CMAKE_SOURCE_DIR` refers to the parent project
(that has nothing to do with capstone), while `PROJECT_SOURCE_DIR` refers
to the root of the capstone source tree.

Recently introduced changes to enable CPack (#2590) are using the wrong
variable and are hence breaking builds that use capstone through
fetch_content. Use the correct variable to fix this issue.

* cmake: Only include cpack in top-level builds

Do not include cpack in builds where capstone is used through
fetch_content.

Rot127 committed:
Update operand type enums of all arch modules to the one in `capstone.h` (#2633)

* Set all operand types to the main CS_OP_ types from capstone.h.

* Add test cases from issue.

Rot127 committed:
Enhance shift value and types of shift instructions. (#2638)

* Enhance shift value and types of shift instructions.

Shifts via registers now save the register id in cs_arch64_op.shift.value
and set the shift type accordingly.

* Sort table

Rot127 committed:
Fix build for compilers requiring explicit static for inline functions.. (#2645)

Rot127 committed:
Tms32c64x Little Endian (#2648)

* Add little endian support for TMS320c64x.
This requires now to initialize TMS320c64x with the
CS_MODE_BIG_ENDIAN flag.

* Fix typo

Rot127 committed:
Add Call group to svc, smc and hvc. (#2651)

hainest committed:
Decode BH field in print_insn_detail_ppc (#2662)

Rot127 commented:
Replaced by https://github.com/capstone-engine/capstone/pull/2670
